src/sentry/discover/endpoints/discover_key_transactions.py

```diff
@@ -0,0 +1,118 @@
+from __future__ import absolute_import
+
+from rest_framework.response import Response
+
+from sentry import features
+from sentry.api.bases import OrganizationEventsV2EndpointBase
+from sentry.models import Project, ProjectStatus
+from sentry.api.bases.organization import OrganizationPermission
+from sentry.api.exceptions import ResourceDoesNotExist
+from sentry.discover.models import KeyTransaction, MAX_KEY_TRANSACTIONS
+from sentry.snuba.discover import query
+
+
+class KeyTransactionEndpoint(OrganizationEventsV2EndpointBase):
+    permission_classes = (OrganizationPermission,)
+
+    def has_feature(self, request, organization):
+        return features.has("organizations:discover", organization, actor=request.user)
+
+    def get_project(self, organization, project_id):
+        """ Get a project for an org, or None if one doesn't exist """
+        return Project.objects.filter(
+            organization=organization, status=ProjectStatus.VISIBLE, id=project_id
+        ).first()
+
+    def post(self, request, organization):
+        """ Create A Key Transaction """
+        if not self.has_feature(request, organization):
+            return self.response(status=404)
+
+        project = self.get_project(organization, int(request.data["project"]))
+
+        if project is None:
+            return Response({"detail": "No project with that id found"}, status=400)
+
+        base_filter = {"organization": organization, "project": project, "owner": request.user}
+
+        # Limit the number of key transactions
+        if KeyTransaction.objects.filter(**base_filter).count() >= MAX_KEY_TRANSACTIONS:
```

Comment:
This won't entirely prevent going over if a user submits concurrent requests. Wrapping this read and the related writes in a transaction will help narrow the window though.